File: code/default_arguments.java
public class Default_arguments {
    public static void main(String[] args) {
        // Java does not support the following named inputs for methods.
        // int result = product(number = 5, multiplier = 3);
        int result = product(5,3);
        // Since there is no default value,
        // the method product uses 5 as the multiplier value instead of 3.

        System.out.println("Das Ergebnis lautet: " + result);
    }

    // Java does not have default parameters.
    public static int product(
            int number,
            int multiplier
    )
    {
        multiplier = 5;
        return number * multiplier;
    }
}
// Ausgabe:
// Das Ergebnis lautet: 25

File: code/default_arguments.kt
fun main() {
    var result = product(zahl = 5) // Default argument multiplikator is used.
    println("Das Ergebnis lautet: $result")
    result = product(5,3) // Default argument multiplikator is not used.
    println("Das Ergebnis lautet: $result")
}

// Function parameters can have default values,
// which are used when you skip the corresponding argument:
fun product(
        zahl: Int,
        multiplikator: Int = 5,
): Int
{
    return zahl * multiplikator
}
//multiplikator is a default value.

// Ausgabe:
// Das Ergebnis lautet: 25
// Das Ergebnis lautet: 15